contracts/strategy/LiquidationStrategy.sol

```diff
@@ -144,80 +162,151 @@ contract LiquidationStrategy is BaseStrategy, ReentrancyGuard, IFlashLendingCall
-            idleWXDC.amountNeededToPayDebt = 0;
+        uint256 receivedAmount = _sellCollateralV2(_collateral, path, _router, _amount, _minAmountOut);
+        
+        emit LogSellCollateralV2(path, _router, _amount, _minAmountOut, dexAmountOut, receivedAmount);
+
+        uint256 balanceOfFXDAfterSwap = fathomStablecoin.balanceOf(address(this));
+        uint256 fxdReceived = balanceOfFXDAfterSwap.sub(balanceOfFXDBeforeSwap); // FXD gained from the swap
+
+        _handleLogicForProfitOrLoss(fxdReceived, costOfCollateralInFXD);
+    }
+
+
+    /// @notice Allows the strategy manager to sell Collateral, held by the contract, to UniV3.
+    /// @dev Only the current strategy manager can call this function.
+    /// This function will work only when the UniversalRouter and Permit2 contracts addresses are already set in uniswapV3Info.
+    /// This allows for managing the Collateral obtained through liquidations.
+    /// @param _universalRouter The UniswapV3(or a fork)'s UniversalRouter  used for the sale.
+    /// @param _amount The amount of Collateral to sell.
+
+    function sellCollateralV3(address _collateral, address _universalRouter, uint256 _amount) external onlyStrategyManager {
+        if (_universalRouter == address(0)) revert ZeroAddress();
+        if (_amount == 0) revert ZeroAmount();
+        if (_amount > idleCollateral[_collateral].collateralAmount) revert WrongAmount();
+        if (uniswapV3Info[_universalRouter].permit2 == address(0) || 
+            uniswapV3Info[_universalRouter].poolFee == 0) revert V3InfoNotSet();
+
+        // Calculate the cost of Collateral in terms of FXD using the average price
+        uint256 _averagePriceOfCollateral = idleCollateral[_collateral].averagePriceOfCollateral;
+        uint256 costOfCollateralInFXD = _amount.mul(_averagePriceOfCollateral).div(WAD); 
+        uint256 balanceOfFXDBeforeSwap = fathomStablecoin.balanceOf(address(this));
+        
+        // Existing logic for selling Collateral
+        idleCollateral[_collateral].collateralAmount -= _amount;
+
+        _nullifyIdleCollateral(_collateral);
+
+        //if idleCollateral is nullified in previous line, no need to _adjustAmountNeededToPayDebt
+        if (idleCollateral[_collateral].collateralAmount != 0) {
+            _adjustAmountNeededToPayDebt(_collateral, _amount.mul(_averagePriceOfCollateral).div(WAD));
         }
 
-        (address[] memory path, uint256 dexAmountOut) = _computeMostProfitablePath(_router, address(WXDC), _amount);
+        uint256 receivedAmount = _sellCollateralV3(_collateral, address(fathomStablecoin), _amount, uniswapV3Info[_universalRouter].poolFee, _universalRouter);
+        emit LogSellCollateralV3(_universalRouter, _amount, receivedAmount); // This line is crucial for logging the sale details
+
+        uint256 balanceOfFXDAfterSwap = fathomStablecoin.balanceOf(address(this));
+        uint256 fxdReceived = balanceOfFXDAfterSwap.sub(balanceOfFXDBeforeSwap); // FXD gained from the swap
```

Comment:
look `uint256 receivedAmount = _sellCollateralV3` calculated as:
```
uint256 _tokencoinBalanceBefore =  ERC20(_tokenOut).balanceOf(address(this));
...
uint256 _tokencoinBalanceAfter = ERC20(_tokenOut).balanceOf(address(this));
receivedAmount = _tokencoinBalanceAfter.sub(_tokencoinBalanceBefore);
```
and here we do precisely the same, and ignore the`receivedAmount`

```
uint256 balanceOfFXDBeforeSwap = fathomStablecoin.balanceOf(address(this));
...
uint256 receivedAmount = _sellCollateralV3(_collateral, address(fathomStablecoin), _amount, uniswapV3Info[_universalRouter].poolFee, _universalRouter);
...
uint256 balanceOfFXDAfterSwap = fathomStablecoin.balanceOf(address(this));sale details

```
this needs to be refactored